File: Structural/Flyweight/Cpp/FlyweightCharacter.cpp
// FlyweightCharacter.cpp

#include "FlyweightCharacter.h"

FlyweightCharacter::FlyweightCharacter(char symbol) : symbol(symbol) {}

void FlyweightCharacter::display() const {
  std::cout << "Character: " << symbol << std::endl;
}

File: Structural/Flyweight/CSharp/FlyweightCharacter.cs
// FlyweightCharacter.cs

using System;

public class FlyweightCharacter {
  private readonly char symbol;

  public FlyweightCharacter(char symbol) {
    this.symbol = symbol;
  }

  public void Display() {
    Console.WriteLine($"Character: {symbol}");
  }
}
